RationaleMCP/0031/grammar.md

```diff
@@ -173,6 +174,24 @@ end _F;
 > ~~_import-list_ → _IDENT_ ( **,** _IDENT_ )*~~
 
 
+## Clock partitions
+
+> _base-partition_ →\
+> &emsp; **partition** _comment_\
+> &emsp; _clock-clause_ **;**\
+> &emsp; _sub-partition_*
+
+> _sub-partition_ →\
+> &emsp; **subpartition** _comment_\
+> &emsp; _clock-clause_ **;**\
+> &emsp; ( **solverMethod** _STRING_ **;** )?\
```

Comment:
We hadn't settled how to specify solver method, so I added this as a starting point.

Reply:
Web Meeting: (Hans, Henrik, Gerd, Martin, Oliver)

Subpartitions do not necessarily require a solver method.
In full Modelica there is no natural place to define a solver method for subpartitions. This makes it look a bit ugly and a risk of conflicting definitions.

Gerd: What I don't like is that the solverMethod looks like a declaration. Maybe we could move it ahead of clock-clause.

Henrik: Keyword 'discretization' would look better as a keyword together with partition/subpartition than the lower camel case 'solverMethod'.
If we swop these two lines then we avoid looking like a declaration.
Dropping the semicolon, but having encourages declaration on a separate line, which improves readability as we had discussed previously.

Oliver: solverMethod is a well established term.

Gerd: Could it remain a named argument?

Henrik: Technically the solverMethod is not associated with the clock but with the subpartition. This could not be expressed in a better way in full Modelica.

Examples:
```
subpartition "Alternative A"
    Clock _subClock0 = subSample('baseClock', 2);
    solverMethod "ImplicitEuler";
    ...
```

```
subpartition "Alternative B"
    solverMethod "ImplicitEuler";
    Clock _subClock0 = subSample('baseClock', 2);
    ...
```

```
subpartition "Alternative C"
    discretization "ImplicitEuler"    annotation(tolerance=1e3);
    Clock _subClock0 = subSample('baseClock', 2);
    ...
```

```
subpartition "Alternative D" discretization "ImplicitEuler" 
    annotation(tolerance=1e3)
    Clock _subClock0 = subSample('baseClock', 2);
    ...
```

```
subpartition(discretization(tolerance=1e3)="ImplicitEuler") "Alternative E" 
    Clock _subClock0 = subSample('baseClock', 2);
    ...
```

```
subpartition subPart1 "Alternative F" 
    extends basepartition(
      discretization = "ImplicitEuler", tolerance=1e3,
      subClock = subSample('baseClock', 2));
    ...
end subPart1;
```

```
subpartition subE2(
  clock=subsample('baseClock',2),
   discretization="ImplicitEuler") "Alternative G"
```

Henrik: There might be the need for more sophisticated attributes for solver methods in the future. This could be done by:
`solverMethod "ImplicitEuler"(tolerance=1e3);`
We have an annotation on the subpartition and could also have an annotation on the solverMethod as well.

Reply:
Web Meeting: Hans, Henrik, Gerd, Martin, Oliver
Revised options above by adding tolerance to all of them.
Added Alternative H.

Discussion to be concluded in next meeting.

Reply:
```
subpartition (
  solverMethod="ImplicitEuler", tolerance=1e-3) "Alternative I" 
  Clock _subClock0 = subsample('baseClock',2);
```